src/tribler/core/upgrade/version_manager.py

```diff
@@ -103,6 +108,9 @@ def __repr__(self):
     def get_directory(self):
         return self.root_state_dir / ('%d.%d' % self.major_minor)
 
+    def get_tmp_copy_directory(self):
+        return self.root_state_dir / ('%d.%d_tmp_copy' % self.major_minor)
```

Comment:
NIT: Maybe it is a bit safer to generate some random postfix for this temporary directory
```suggestion
        major = self.major_minor[0]
        minor = self.major_minor[1]
        temporary_dir_name = f'{major}.{minor}_tmp_copy_{random.randrange(100)}'
        return self.root_state_dir / temporary_dir_name
```